migrating_gazebo_classic_ros2_packages.md

```diff
@@ -0,0 +1,593 @@
+for details on each parameter. Here's the full `<plugin>` tag with comments
+describing the mapping from the original plugin.
+
+```xml
+<plugin filename="gz-sim-diff-drive-system" name="gz::sim::systems::DiffDrive">
+  <!-- Remove <ros> tag. -->
+
+  <!-- wheels -->
+  <left_joint>wheel_left_joint</left_joint>
+  <right_joint>wheel_right_joint</right_joint>
+
+  <!-- kinematics -->
+  <wheel_separation>0.287</wheel_separation>
+  <wheel_radius>0.033</wheel_radius> <!-- computed from <wheel_diameter> in the original plugin-->
+
+  <!-- limits -->
+  <max_linear_acceleration>0.033</max_linear_acceleration> <!-- computed from <max_linear_acceleration> in the original plugin-->
+
+  <topic>cmd_vel</topic> <!-- from <commant_topic> -->
+
+  <odom_topic>odom</odom_topic> <!-- from <odometry_topic> -->
+  <frame_id>odom</frame_id> <!-- from <odometry_frame> -->
+  <child_frame_id>base_footprint</child_frame_id> <!-- from <robot_base_frame> -->
+  <odom_publisher_frequency>30</odom_publisher_frequency> <!-- from <update_rate>-->
+
+  <tf_topic>/tf</tf_topic> <!-- Short topic name for tf output -->
+
+</plugin>
+
+```
+
+The `<wheel_torque>` parameter can be realized by setting effort limits on each
+`<joint>`. For example:
+
+```xml
+
+<joint name="wheel_right_joint" type="revolute">
+  <parent>base_link</parent>
+  <child>wheel_right_link</child>
+  <pose>0.0 -0.144 0.023 -1.57 0 0</pose>
+  <axis>
+    <xyz>0 0 1</xyz>
+    <limit>
+      <effort>20</effort> <!-- from <wheel_torque> in libgazebo_ros_diff_drive.so.-->
+    </limit>
+  </axis>
+</joint>
+```
+
+### libgazebo_ros_joint_state_publisher.so
+
+We will replace this plugin as well with
+[`JointStatePublisher`](https://gazebosim.org/api/gazebo/6/classignition_1_1gazebo_1_1systems_1_1JointStatePublisher.html).
+The parameters are mostly similar, however, the `<update_rate>` parameter is not
+supported. Here's the full `<plugin>` tag with comments describing the mapping
+from the original plugin.
+
+```xml
+<plugin filename="gz-sim-joint-state-publisher-system"
+  name="gz::sim::systems::JointStatePublisher">
+  <topic>joint_states</topic> <!--from <ros><remaping> -->
+  <joint_name>wheel_left_joint</joint_name>
+  <joint_name>wheel_right_joint</joint_name>
+</plugin>
+```
+
+### World plugins
+
+As mentioned earlier, sensors are handled by generic world level plugins. By
+default, if there are no world plugins specified, Gazebo adds the `Physics`,
+`SceneBroadcaster`, and `UserCommands` plugins by default. However, if we
+specify any plugins at all, Gazebo will assume we want to override the defaults,
+so will not add any default plugins. Therefore, we have to add the additional
+plugins for IMU and Lidar sensors as well as the ones that would have been added
+by default. We will once again edit `turtlebot3_gazebo/worlds/empty_world.world`
+add the following right after `<world name="default">`.
```

Comment:
I understand this behavior, but we may want  to emphasize this section as it is a bit counter intuitive.

Reply:
What do you think about this:

![image](https://github.com/gazebosim/docs/assets/206116/b0ae8a0f-89c0-406b-9cfe-04e3362328fb)